apache/netbeans
# Add support for Jakarta EE 11

This PR should add almost full support for Jakarta EE 11 (with the same support that we have for Jakarta EE 10), there are missing pieces like:
- The new schemas, I used new schemas from from [master](https://github.com/jakartaee/jakartaee-schemas/tree/master/xml/src) and create the missing ones with the same content as the Jakarta EE 10 ones (I will replace them with final versions)
- There is no new logic for new specifications like [Jakarta Data](https://github.com/jakartaee/data)
- Some maven and gradle coordinates does not exist yet including the one from @juneau001
- Something that I forgot or broke

In order to test this PR you will need:
- [GlassFish 8](https://github.com/eclipse-ee4j/glassfish/releases), add the new schemas into `/glassfish/lib/schemas`
- Add the new Jakarta 11 jars into `/glassfish/modules/` (e.g. [persistence 3.2](https://central.sonatype.com/artifact/jakarta.persistence/jakarta.persistence-api))
- Use Ant projects.

These steps will decrease with upcoming GlassFish/Tomcat milestones and, new and updated schemas.

Jakarta EE 11 will be rolling out in [waves](https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan) and the changes needed for NetBeans should be minimal. I will add support for new GlassFish 8 and Tomcat/TomEE versions the moment they are available.

NetBeans Platform changes:

- Add support for GlassFish 8
- Add Jakarta EE 11 deployment descriptors and schemas from [master](https://github.com/jakartaee/jakartaee-schemas/tree/master/xml/src)
- Add core support for Jakarta EE 11
  - CDI 4.1
  - JSP 4.0
  - Servlet 6.1
  - Validation 3.1 (with Constraint 3.1)
  - JSF 4.1
  - Persistence 3.2
- Add initial support for Jakarta EE 11 with maven (missing coordinates)
- Add initial support for Jakarta EE 11 with gradle (missing coordinates)
- Add support for Jakarta EE 11 with ant 
- Add initial support for Jakarta EE 11 with Tomcat and TomEE
- Add initial support for Jakarta EE 11 with Payara
- Add initial support for Jakarta EE 11 with WildFly
- Add three new methods to `Profile.class`: `isWebProfile()`,  `isFullProfile()`, and `isAtMost()`
- Add new method `isAtMost()` to `JsfVersion.class`
- Add new NetBeans modules `jakartaee11.api` and `jakartaee11.platform`
- Add some missing Jakarta EE 10 properties, schemas, actions, and logic

NetBeans Testing:

- Verify successful execution of libraries and licenses Ant test
- Verify successful execution of Verify Sigtests
- Verify successful execution of unit tests for modules `glassfish.common`, `glassfish.javaee`, `glassfish.tooloing`, and `glassfish.eecommon`
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Successfully register GlassFish 8.0.0, create a web app, EAR, App Client, EJB and verify (all with Ant):
  - Creation of deployment descriptors (web, ejb, beans, jpa, and ear)
  - Clean, build and run the projects
